app/models/filesystem.rb

```diff
@@ -120,6 +120,15 @@ def contents_displayable?
     !mime_type.binary?
   end
 
+  def displayable_contents
+    if contents_displayable? &&
+       ((contents[0].ord == 255 && contents[1].ord == 254) || (contents[0].ord == 254 && contents[1].ord == 255))
```

Comment:
How are the `contents` in US-ASCII?  Does it comes that way from the database?  I'm partly asking because if that's the case, your test doesn't really test that.  I would expect the test to have `let(:filesystem_file_utf16_little_endian) { "\xff\xfe\r\x00\n\x00".force_encoding("US-ASCII") }`

I guess `contents` could be both depending on if you were to check this value before or after saving to the database.  Note that you might be able to test this with `FactoryGirl.create` followed by `FileSystem.find` (`contents` retrieved from the database), and then a separate test with `FactoryGirl.build` (`contents` accessed immediately after setting it but *not* going to the database).

In either case, I think the numerical approach you have is nicest, given the different encodings, but might be cleaner using the `bytes` method...

```ruby
UTF_16BE_BOM = "\xFE\xFF".bytes.freeze # Or [254, 255].freeze
UTF_16LE_BOM = "\xFF\xFE".bytes.freeze # Or [255, 254].freeze
# ...
bom = contents.each_byte.take(2) # Or contents.byteslice(0, 2).bytes
bom == UTF_16BE_BOM || bom == UTF_16LE_BOM
```

or

```ruby
UTF_16_BOMS = [UTF_16BE_BOM, UTF_16LE_BOM]
# ...
UTF_16_BOMS.include?(bom)
```